src/extensions/default/HealthData/unittests.js

```diff
@@ -0,0 +1,126 @@
+            var ONE_DAY = 24 * 60 * 60 * 1000,
+                prefs,
+                HealthDataManager;
+            
+            beforeEach(function () {
+                PreferencesManager = testWindow.brackets.test.PreferencesManager;
+                prefs = PreferencesManager.getExtensionPrefs("healthData");
+                HealthDataManager = testWindow.brackets.test.HealthDataManager;
+            });
+            
+            afterEach(function () {
+                HealthDataManager = null;
+                prefs = null;
+                PreferencesManager = null;
+            });
+
+            it("should send data to server", function () {
+                PreferencesManager.setViewState("lastTimeSendHealthData", (new Date()).getTime() - ONE_DAY);
+                PreferencesManager.setViewState("healthDataNotificationShown", true);
+                var promise = HealthDataManager.checkHealthDataSend();
+                waitsForDone(promise, "Send Data to Server", 4000);
+            });
+
+            it("should not send data to server", function () {
+                PreferencesManager.setViewState("lastTimeSendHealthData", (new Date()).getTime() - ONE_DAY);
+                prefs.set("healthDataTracking", false);
+                var promise = HealthDataManager.checkHealthDataSend();
+                waitsForFail(promise, "Send Data to Server", 4000);
+            });
+
+        });
+        
+        describe("Notification is displayed", function () {
+            var HealthDataNotification;
+            
+            beforeEach(function () {
+                HealthDataNotification = testWindow.brackets.test.HealthDataNotification;
+            });
+            
+            afterEach(function () {
+                HealthDataNotification = null;
+            });
+            
+            it("should show notification dialog", function () {
+                HealthDataNotification.showDialogHealthDataNotification();
+                expect($(testWindow.document).find(".health-data-notification.instance").length).toBe(1);
+            });
+        });
+
+        describe("Health Data Statistics is displayed", function () {
+            var HealthDataPreview;
+            
+            beforeEach(function () {
+                HealthDataPreview = testWindow.brackets.test.HealthDataPreview;
+            });
+            
+            afterEach(function () {
+                HealthDataPreview = null;
+            });
+            
+            it("should show file preview dialog", function () {
+
+                runs(function () {
+                    var promise = HealthDataPreview.previewHealthData();
+                    waitsForDone(promise, "Health Data File Preview", 4000);
+                });
+
+                runs(function () {
+                    expect($(testWindow.document).find(".health-data-preview.instance").length).toBe(1);
+                });
+
+            });
+        });
```

Comment:
It would be great to add some stronger tests here in the future:

* Ensure the first data-send is not triggered until 24 hours after first launch (you'd have to parameterize the timeout value so tests don't take 1 day to run, though :-)
* Ensure no data is sent if the opt-out preference is set
* Ensure an extension id that's not in the registry isn't returned by HealthDataUtils